src/ngraph/runtime/gpu/cuda_emitter.cpp

```diff
@@ -676,6 +680,236 @@ size_t runtime::gpu::CUDAEmitter::build_elementwise_n_to_1(const GPURuntimeConte
+                }
+                writer.block_end();
+                writer << "out[tid] = in_bounds ? source[source_idx] : in[tid];\n";
+            }
+            writer.block_end();
+        }
+        writer.block_end();
+        compiled_kernel = ctx->compiled_kernel_pool->set(kernel_name, writer.get_code());
+    }
+
+    std::vector<int> params;
+    params.reserve(tensor_shape.size() * 3 - 1 + lower_bounds.size() + upper_bounds.size() +
+                   slice_strides.size() * 3 + source_shape.size() * 2);
+
+    int dstr_offset = params.size();
+    auto dim_strides = row_major_strides(tensor_shape);
+    for (int i = 0; i < dim_strides.size() - 1; i++)
+    {
+        params.push_back(static_cast<int>(dim_strides[i]));
+    }
+    int dmagic_offset = params.size();
+    std::vector<int> dmagics;
+    std::vector<int> dshifts;
+    for (int i = 0; i < tensor_shape.size(); i++)
+    {
+        int magic;
+        int shift;
+        std::tie(magic, shift) = idiv_magic_u64(static_cast<int>(dim_strides[i]));
+        dmagics.push_back(magic);
+        dshifts.push_back(shift);
+    }
+    std::copy(dmagics.begin(), dmagics.end(), std::back_inserter(params));
+    int dshift_offset = params.size();
+    std::copy(dshifts.begin(), dshifts.end(), std::back_inserter(params));
+    int lbound_offset = params.size();
+    for (auto const& lbound : lower_bounds)
+    {
+        params.push_back(static_cast<int>(lbound));
+    }
+    int ubound_offset = params.size();
+    for (auto const& ubound : upper_bounds)
+    {
+        params.push_back(static_cast<int>(ubound));
+    }
+    int slice_str_offset = params.size();
+    for (auto const& slstr : slice_strides)
+    {
+        params.push_back(static_cast<int>(slstr));
+    }
+    int slice_magic_offset = params.size();
+    std::vector<int> smagics;
+    std::vector<int> sshifts;
+    for (int i = 0; i < slice_strides.size(); i++)
+    {
+        int magic;
+        int shift;
+        std::tie(magic, shift) = idiv_magic_u64(static_cast<int>(slice_strides[i]));
+        smagics.push_back(magic);
+        sshifts.push_back(shift);
+    }
+    std::copy(smagics.begin(), smagics.end(), std::back_inserter(params));
+    int slice_shift_offset = params.size();
+    std::copy(sshifts.begin(), sshifts.end(), std::back_inserter(params));
+    int source_shape_offset = params.size();
+    for (auto const& dim : source_shape)
+    {
+        params.push_back(static_cast<int>(dim));
+    }
+    int source_strides_offset = params.size();
+    auto source_strides = row_major_strides(source_shape);
+    for (auto const& str : source_strides)
+    {
+        params.push_back(static_cast<int>(str));
+    }
+
+    size_t nthreads = shape_size(tensor_shape);
+    int rank = tensor_shape.size();
+
+    // TODO: blending factors are not currently implemented
+    float alpha = 1.0f;
+    float beta = 0.0f;
+
+    std::unique_ptr<gpu::primitive> replace_slice(new gpu::primitive{[=](void** inputs,
```

Comment:
The lambda is very large here and embedding it in the function call really hides things. Define the lambda first then use this new variable to build the replace_slice